### PR TITLE
Dev 6.0.0

### PR DESCRIPTION
**Pull request by hkazuakey**
## What the difference from the previous version.

- Work with Solr 6.0.0
Work with the latest version of Solr / Lucene

- IPA dictionary version is updated to 2.6.1.

- Introducing compatibilityMode (default is true)
It enables us to control how to handle Katakana UNKNOWN tokens. If the parameter is set to true, it works as before that concatenate concecutive Katakana tokens when a token is UNKNOWN. On the other hand, if it's set to false, it prevents generating one UNKNOWN token. Ideally, deep decomposition feature or another intelligent mechanism that splits Katakana text into tokens should be implemented but it would need more time, thus, for the time being, this parameter will help for people who want to change this behaviour. 

## 前バージョンからの変更点
- Solr 6.0.0 をサポートしました。

- IPA辞書のバージョンを2.6.1へ変更しました。

- compatibilityMode というパラメートを追加しました（デフォルトではtrue)
以前のlucene-gosenでは、カタカナの入力に対して、未知語があった場合に1語の未知語（UNKNOWN）としていましたが、この挙動を変更できるようにしました。
理想としては、Deep decomposition 機能など、カタカナ未知語の混ざったカタカナテキストでも、分かち書きをする機能を入れた方が良いと思いますが、それにはもう少し時間が必要なので、まずは１語にしないという選択肢が得られるようにしました。

### CODE REVIEW

**Comment by johtani** [comment]
Can you squash these commits?

**Comment by hkazuakey** [comment]
I've finished squashing.

File: src/java/net/java/sen/SenFactory.java [hunk 1]
```diff
@@ -125,10 +125,10 @@ private static InputStream getInputStream(String name, String dictionaryDir) thr
 
   private final String[] posIndex, conjTypeIndex, conjFormIndex;
   private final ByteBuffer costs, pos, tokens, trie;
-  
-  
+
   public static final String unknownPOS = "未知語";
-  
+  public static boolean compatibilityMode = true;
```
Comment (johtani) [inline src/java/net/java/sen/SenFactory.java]: "compatibilityMode" is affect only Unknown Katakana token, right?
I think we should change "compatibilityMode" to "tokenizeUnknownKatakana" or something else.

Comment (johtani) [inline src/java/net/java/sen/SenFactory.java]: And this property should not be static property. If we set it once and we call getStringTagger(), its behavior is different before.
I think we change it to instance property and we add 2nd param to each method.

**Comment by johtani** [comment]
@hkazuakey Thanks for your PR.
I left some comments.

**Comment by hkazuakey** [comment]
@johtani Thank you for your review and comments!
I will update my code.

**Comment by hkazuakey** [comment]
I've pushed the latest changes that follows your advice. 
The flag name is changed to tokenizeUnknownKatakana and moved to the class property of Tokenizer class, which enables us to create multiple tokenizers with different configurations of this flag.

File: src/java/org/apache/lucene/analysis/gosen/GosenTokenizer.java [hunk 2]
```diff
@@ -74,24 +74,27 @@
   // so we accumulate this so we can then subtract to present an absolute cost.
   private int accumulatedCost = 0;
 
+  /**
+   * Constructors
+   */
   public GosenTokenizer() {
-    this(DEFAULT_TOKEN_ATTRIBUTE_FACTORY, null, null);
+    this(DEFAULT_TOKEN_ATTRIBUTE_FACTORY, null, null, false);
```
Comment (johtani) [inline src/java/org/apache/lucene/analysis/gosen/GosenTokenizer.java]: Could you move default value "false" to a constant?

**Comment by johtani** [comment]
I left a comment, but LGTM. Thanks a lot of work!

**Comment by hkazuakey** [comment]
Sorry for this late response. I have modified GosenTokenizer and Analyzer to introduce DEFAULT_UNKNOWN_KATAKANA_TOKENIZATION which holds default value for the new flag that controls Unknown Katakana Tokenization behavior.

File: src/java/org/apache/lucene/analysis/gosen/GosenAnalyzer.java [hunk 3]
```diff
@@ -106,9 +108,15 @@ public GosenAnalyzer(CharArraySet stopwords, Set<String> stoptags, CharArraySet
    * @param stoptags          a stoptags set: words containing these parts of speech will be removed from the stream.
    * @param stemExclusionSet  a stemming exclusion set: these words are ignored by
    *                           {@link GosenBasicFormFilter} and {@link GosenKatakanaStemFilter}
+<<<<<<< HEAD
```
Comment (johtani) [inline src/java/org/apache/lucene/analysis/gosen/GosenAnalyzer.java]: Something wrong...

**Comment by johtani** [comment]
Thanks, I leave a comment.
LGTM if you fix it.

File: src/java/org/apache/lucene/analysis/gosen/GosenAnalyzer.java (hunk 3 above)
Comment (hkazuakey) [inline src/java/org/apache/lucene/analysis/gosen/GosenAnalyzer.java]: It seems I have pushed wrong file...

**Comment by hkazuakey** [comment]
Fixed. Thank you!

**Comment by johtani** [comment]
@hkazuakey Thanks! Could you merge this to master branch?

**Pull request merged**